docs/requirements.txt

```
@@ -0,0 +1,12 @@
+# Required libraries
+
+# Example
+scipy>=1.9.3
+numpy>=1.19.0
+pandas>=1.4.0
+tqdm>=4.50.2
+caf.toolkit>=0.2.1
+sphinx-automodapi>=0.16.0
+pydata-sphinx-theme>=0.14.1
+caf.distribute>=0.1.0
+graphviz>=0.20.1
```

Can this be as simple as:
```suggestion
caf.distribute>=0.1.0
sphinx-automodapi>=0.16.0
pydata-sphinx-theme>=0.14.1
graphviz>=0.20.1
```

Making a few assumptions here:
- These requirements are just to build the docs
- only distribute is really needed so that it can get the version

---

All are needed for the sphinx build github action. Although all other requirements are installed by installing distribute, I think it's better practice to keep them all in explicitly but it's your call.

---

Do you know how the sphinx build installs distribute? Is it a local install?

If so, it will just use the already defined requirements.txt. At least this way we prevent repeating ourselves - we are only copying that file into here after all.

---

I don't think it's local no, it's from pypi, so as an edge case it could install a previous version with different requirements and then raise an error when it tries to import something which isn't installed. It isn't an issue if the version of distribute is out of date because it just needs to be able to import it without an error message. As I think about it that same error could occur on readthedocs too.